buffer/src/main/java/io/netty/buffer/ReadOnlyByteBufferBuf.java

```diff
@@ -195,11 +195,6 @@ public ByteBuf getBytes(int index, ByteBuf dst, int dstIndex, int length) {
     public ByteBuf getBytes(int index, byte[] dst, int dstIndex, int length) {
         checkDstIndex(index, length, dstIndex, dst.length);
 
-        if (dstIndex < 0 || dstIndex > dst.length - length) {
```

Comment:
imho the exception message is lot more useful here... so not sure we really should remove this.

Reply:
@normanmaurer presumably you mean more useful since it will be `dstIndex:` rather than `index:`. The main issues imo are the duplicate check and inconsistency - `chkDstIndex(...)` is used equivalently in numerous other places without the additional check following it.

Actually looking again I think there's a small change we can make which will give of both worlds, I'll add a commit.

Reply:
@normanmaurer I changed `chkSrcIndex` and `chkDstIndex` in `AbstractByteBuf` to have a more specific exception message, so now everywhere will benefit.

Also I just realized that the removed check here is really useless in any case - it will never evaluate to true since `chkDskIndex` above would already have thrown. So you would never see the more useful message.